mpqp/noise/noise_model.py

```diff
@@ -1,24 +1,31 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
+from functools import reduce
+from itertools import product
 from typing import TYPE_CHECKING, Optional, Sequence
 
+import numpy as np
+import numpy.typing as npt
+
+from mpqp.tools.generics import T
+from mpqp.tools.maths import Id, pauli_X, pauli_Y, pauli_Z
+
 if TYPE_CHECKING:
     from braket.circuits.noises import Noise as BraketNoise
     from braket.circuits.noises import TwoQubitDepolarizing
     from qat.quops.class_concepts import QuantumChannel as QLMNoise
     from qiskit_aer.noise.errors.quantum_error import QuantumError
 
-import numpy as np
 from typeguard import typechecked
 
 from mpqp.core.instruction.gates.native_gates import NativeGate
 from mpqp.core.languages import Language
-from mpqp.noise.custom_noise import KrausRepresentation
 from mpqp.tools.generics import T
 
 
 def plural_marker(items: Sequence[T]):
```

Comment:
I've added documantation but I think this is a pretty niche function, so I've
set it to be a private function instead of putting it in tools. Tell me if you
agree with this choice.

Reply:
Ok to put it private